### PR TITLE
UkrFormalityClassification added

### PR DESCRIPTION
**Pull request by dokato**
Translated English GYAFC data to Ukrainian: https://huggingface.co/datasets/ukr-detect/ukr-formality-dataset-translated-gyafc .

## Checklist for adding MMTEB dataset

Reason for dataset addition:

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [x] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

File: mteb/tasks/Classification/ukr/UkrFormalityClassification.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        n_samples={"test": 4853},
+        avg_character_length={"test": 52.39},
+    )
+
+    def dataset_transform(self):
+        self.dataset = self.dataset.rename_column("labels", "label")
+        self.dataset = self.dataset.class_encode_column("label")
+        self.dataset = self.dataset["test"].train_test_split(
+            test_size=0.5, seed=self.seed, stratify_by_column="label"
+        )  # balanced sampling
```
Comment (imenelydiaker) [inline mteb/tasks/Classification/ukr/UkrFormalityClassification.py:55]: You can use our helper function like this:
```suggestion
        self.dataset = self.stratified_subsampling(self.dataset, seed=self.seef, splits=["test")
```

File: mteb/tasks/Classification/ukr/UkrFormalityClassification.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+if __name__ == "__main__":
+    from sentence_transformers import SentenceTransformer
+
+    from mteb import MTEB
+
+    # intfloat/multilingual-e5-small
+    model_name = "intfloat/multilingual-e5-small"
+    model = SentenceTransformer(model_name)
+    evaluation = MTEB(tasks=[UkrFormalityClassification()])
+    evaluation.run(model)
```
Comment (imenelydiaker) [inline mteb/tasks/Classification/ukr/UkrFormalityClassification.py:67]: You can also remove this before merging

**Review by imenelydiaker — COMMENTED** [review]
LGTM, I'll let you address my comments and add points then we'll be good to merge 🙂

File: mteb/tasks/Classification/ukr/UkrFormalityClassification.py (hunk 1 above)
Comment (dokato) [reply mteb/tasks/Classification/ukr/UkrFormalityClassification.py:55]: I don't think it does the same. I only stratifies "test", leaving large train and validation intact, whereas here, I split the "test" subset into train and test to reduce the overall sample size.

**Comment by dokato** [comment]
Should it be +4 for language in general or lang in task @imenelydiaker ? Also, doe we have some convenient way to list of all lang per task?

**Comment by imenelydiaker** [comment]
> Also, doe we have some convenient way to list of all lang per task?

I'm not sure what you mean here?

File: mteb/tasks/Classification/ukr/UkrFormalityClassification.py (hunk 1 above)
Comment (imenelydiaker) [reply mteb/tasks/Classification/ukr/UkrFormalityClassification.py:55]: Okay I see, I tend to not add datasets where the train set is not provided, but I see what you mean. Following your comment on PR #575, @isaac-chung so for tasks where the dataset does not contain a train split, we should allow creating it from the test set?

**Comment by dokato** [comment]
> I'm not sure what you mean here?

Like a function that crawls through all tasks and lists unique language for Classification, Retrieval etc. I think as the number of datasets grow, it'd be convenient.

**Comment by imenelydiaker** [comment]
> > I'm not sure what you mean here?
> 
> Like a function that crawls through all tasks and lists unique language for Classification, Retrieval etc. I think as the number of datasets grow, it'd be convenient.

I'm not sure we have this implemented, @KennethEnevoldsen may know if we already have this? Else, feel free to open a PR with the proposed implementation

File: mteb/tasks/Classification/ukr/UkrFormalityClassification.py (hunk 1 above)
Comment (isaac-chung) [reply mteb/tasks/Classification/ukr/UkrFormalityClassification.py:55]: I think we can allow it, as long as we state that clearly somewhere, like "For datasets without train splits, we create a train split (by doing XYZ)". I'm curious if @KennethEnevoldsen has an opinion on this too.

Comment (dokato) [reply mteb/tasks/Classification/ukr/UkrFormalityClassification.py]: Ok, sorry, I overlooked. This actually has training and val split, so I made it with `stratified_subsampling` as requested.

**Comment by imenelydiaker** [comment]
@dokato please update your branch and add points and we'll be good to merge 🙂
